### PR TITLE
Configure filebeat keystore

### PR DESCRIPTION
**Pull request by arturo-seijas**
Applicable spec: <link>

### Overview

Configure filebeat keystore for proper authentication with the indexer

### Rationale

N/A

### Juju Events Changes

N/A

### Module Changes

charm.py
wazuh.py
state.py

### Library Changes

N/A

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: src/state.py [hunk 1]
```diff
@@ -135,6 +144,10 @@ def from_charm(
             InvalidStateError: if the state is invalid.
         """
         try:
+            secret_id = indexer_relation_data.get("secret-user")
+            secret_content = charm.model.get_secret(id=secret_id).get_content()
```
Comment (amandahla) [inline src/state.py:148]: if secret not found?

File: src/wazuh.py [hunk 2]
```diff
@@ -204,3 +204,44 @@ def pull_configuration_files(container: ops.Container) -> None:
...
+                "username",
+                "--stdin",
+                "--force",
+            ]
+        )
+        process.wait_output()
+        process = container.exec(
+            [
+                "echo",
+                password,
```
Comment (amandahla) [inline src/wazuh.py:235]: Could password contains special chars?

File: src/state.py (hunk 1 above)
Comment (arturo-seijas) [reply src/state.py:148]: I validation error is raised. This is within a try-except block

File: src/wazuh.py (hunk 2 above)
Comment (arturo-seijas) [reply src/wazuh.py:235]: From what I could gather, they are alphanumeric

**Review by amandahla — APPROVED** [review]

**Review by amandahla — APPROVED** [review]

**Review by gregory-schiano — APPROVED** [review]

**Pull request merged**